core/src/main/java/com/google/instrumentation/stats/Distribution.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.stats;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * A distribution contains summary statistics for a population of values and, optionally, a
+ * histogram representing the distribution of those values across a specified set of histogram
+ * buckets.
+ *
+ * <p>The bucket boundaries for that histogram are described by {@link BucketBoundaries}, which
+ * defines {@code BucketBoundaries.getList().size() + 1 (= N)} buckets.
+ * The boundaries for bucket index i are:
+ *
+ * <ul>
+ *   <li>[-infinity, bounds[i]) for i == 0
+ *   <li>[bounds[i-1], bounds[i]) for 0 &lt; i &lt; N-1
+ *   <li>[bounds[i-1], +infinity) for i == N-1
+ * </ul>
+ *
+ * <p>i.e. an underflow bucket (number 0), zero or more finite buckets (1 through N - 2, and an
+ * overflow bucket (N - 1), with inclusive lower bounds and exclusive upper bounds.
+ *
+ * <p>Note: If N = 1, there are no finite buckets and the single bucket is both the overflow and
+ * underflow bucket.
```

Comment:
Could you please add a TODO to determine whether N = 1 is valid?  Then we should make this comment consistent with the validation of the bucket boundaries.